framework/messages/fr/yii.php

```diff
@@ -17,30 +17,16 @@
-    '{nFormatted} B' => '{nFormatted} o',
-    '{nFormatted} GB' => '{nFormatted} Go',
-    '{nFormatted} GiB' => '{nFormatted} Gio',
-    '{nFormatted} KB' => '{nFormatted} Ko',
-    '{nFormatted} KiB' => '{nFormatted} Kio',
-    '{nFormatted} MB' => '{nFormatted} Mo',
-    '{nFormatted} MiB' => '{nFormatted} Mio',
-    '{nFormatted} PB' => '{nFormatted} Po',
-    '{nFormatted} PiB' => '{nFormatted} Pio',
-    '{nFormatted} TB' => '{nFormatted} To',
-    '{nFormatted} TiB' => '{nFormatted} Tio',
-    '{nFormatted} {n, plural, =1{byte} other{bytes}}' => '{nFormatted} {n, plural, =1{octet} other{octets}}',
-    '{nFormatted} {n, plural, =1{gibibyte} other{gibibytes}}' => '{nFormatted} {n, plural, =1{# gigaoctet} other{# gigaoctets}}',
-    '{nFormatted} {n, plural, =1{gigabyte} other{gigabytes}}' => '{nFormatted} {n, plural, =1{gibioctet} other{gibioctets}}',
-    '{nFormatted} {n, plural, =1{kibibyte} other{kibibytes}}' => '{nFormatted} {n, plural, =1{kibioctet} other{kibioctets}}',
-    '{nFormatted} {n, plural, =1{kilobyte} other{kilobytes}}' => '{nFormatted} {n, plural, =1{# kilooctet} other{# kilooctets}}',
-    '{nFormatted} {n, plural, =1{mebibyte} other{mebibytes}}' => '{nFormatted} {n, plural, =1{mebioctet} other{mebioctets}}',
-    '{nFormatted} {n, plural, =1{megabyte} other{megabytes}}' => '{nFormatted} {n, plural, =1{# megaoctet} other{# megaoctets}}',
-    '{nFormatted} {n, plural, =1{pebibyte} other{pebibytes}}' => '{nFormatted} {n, plural, =1{pebioctet} other{pebioctets}}',
-    '{nFormatted} {n, plural, =1{petabyte} other{petabytes}}' => '{nFormatted} {n, plural, =1{# petaoctet} other{# petaoctets}}',
-    '{nFormatted} {n, plural, =1{tebibyte} other{tebibytes}}' => '{nFormatted} {n, plural, =1{# teraoctet} other{# teraoctets}}',
-    '{nFormatted} {n, plural, =1{terabyte} other{terabytes}}' => '{nFormatted} {n, plural, =1{# teraoctet} other{# teraoctets}}',
+    'No help for unknown command "{command}".' => '@@Aucune aide disponible pour la commande inconnue : "{command}".@@',
+    'No help for unknown sub-command "{command}".' => '@@Aucune aide disponible pour la sous-commande inconnue : "{command}".@@',
+    'Unknown command "{command}".' => '@@Commande inconnue : "{command}".@@',
+    '{attribute} contains wrong subnet mask.' => '@@{attribute} contient un masque de sous-réseau non valide.@@',
+    '{attribute} is not in the allowed range.' => '@@{attribute} n\'est pas dans la plage autorisée.@@',
+    '{attribute} must be a valid IP address.' => '@@{attribute} doit être une adresse IP valide.@@',
+    '{attribute} must be an IP address with specified subnet.' => '@@{attribute} doit être une adresse IP avec un sous-réseau.@@',
+    '{attribute} must not be a subnet.' => '@@{attribute} ne doit pas être un sous-réseau.@@',
+    '{attribute} must not be an IPv4 address.' => '@@{attribute} ne doit pas être une adresse IPv4.@@',
+    '{attribute} must not be an IPv6 address.' => '@@{attribute} ne doit pas être une adresse IPv6.@@',
```

Comment:
The thing here is that the error messages in IpValidator are not explicitly
surrounded by Yii::t() calls, but dynamically translated in
IpValidator::clientValidateAttribute(), so the messages extraction scripts
marks them as unused, even if they are.

Thought about setting up all messages in the init() method using Yii::t()
but I'm still a bit PR shy.. should I go ahead a do it?



On Wed, Oct 28, 2015 at 11:14 PM, Carsten Brandt <notifications@github.com>
wrote:

> In framework/messages/fr/yii.php
> <https://github.com/yiisoft/yii2/pull/9988#discussion_r43325713>:
>
> > -    '{nFormatted} {n, plural, =1{mebibyte} other{mebibytes}}' => '{nFormatted} {n, plural, =1{mebioctet} other{mebioctets}}',
> > -    '{nFormatted} {n, plural, =1{megabyte} other{megabytes}}' => '{nFormatted} {n, plural, =1{# megaoctet} other{# megaoctets}}',
> > -    '{nFormatted} {n, plural, =1{pebibyte} other{pebibytes}}' => '{nFormatted} {n, plural, =1{pebioctet} other{pebioctets}}',
> > -    '{nFormatted} {n, plural, =1{petabyte} other{petabytes}}' => '{nFormatted} {n, plural, =1{# petaoctet} other{# petaoctets}}',
> > -    '{nFormatted} {n, plural, =1{tebibyte} other{tebibytes}}' => '{nFormatted} {n, plural, =1{# teraoctet} other{# teraoctets}}',
> > -    '{nFormatted} {n, plural, =1{terabyte} other{terabytes}}' => '{nFormatted} {n, plural, =1{# teraoctet} other{# teraoctets}}',
> > +    'No help for unknown command "{command}".' => '@@Aucune aide disponible pour la commande inconnue : "{command}".@@',
> > +    'No help for unknown sub-command "{command}".' => '@@Aucune aide disponible pour la sous-commande inconnue : "{command}".@@',
> > +    'Unknown command "{command}".' => '@@Commande inconnue : "{command}".@@',
> > +    '{attribute} contains wrong subnet mask.' => '@@{attribute} contient un masque de sous-réseau non valide.@@',
> > +    '{attribute} is not in the allowed range.' => '@@{attribute} n\'est pas dans la plage autorisée.@@',
> > +    '{attribute} must be a valid IP address.' => '@@{attribute} doit être une adresse IP valide.@@',
> > +    '{attribute} must be an IP address with specified subnet.' => '@@{attribute} doit être une adresse IP avec un sous-réseau.@@',
> > +    '{attribute} must not be a subnet.' => '@@{attribute} ne doit pas être un sous-réseau.@@',
> > +    '{attribute} must not be an IPv4 address.' => '@@{attribute} ne doit pas être une adresse IPv4.@@',
> > +    '{attribute} must not be an IPv6 address.' => '@@{attribute} ne doit pas être une adresse IPv6.@@',
>
> messages with @@ should be removed as they are not found in framework
> code anymore, for IP validator I am not sure what happened here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yiisoft/yii2/pull/9988/files#r43325713>.
>



-- 
Mehdi Achour